ntbs-service/DataAccess/PostcodeRepository.cs

```diff
@@ -0,0 +1,26 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ntbs_service.Models;
+
+namespace ntbs_service.DataAccess
+{
+    public interface IPostcodeRepository
+    {
+        Task<PostcodeLookup> FindPostcode(string postcode);
+    }
+
+    public class PostcodeRepository : IPostcodeRepository

```

Comment:
Is this class providing much value?
Should the PostcodeService have access to `context` directly and do the lookup itself?

Reply:
Yeah when I started the repository pattern I wasn't sure whether we should add one for every type - it makes sense for notification, as there are quite a few, but maybe otherwise best to observe it into the service as Jan suggested

Reply:
I was thinking repository as a way to access data only. and then service prepares parameters or data before and/or after data is retrieved by repository. But can remove repository and access data in service, what do you think?

Reply:
I agree in general, but I can't see this postcodeRepository ever growing, and hence we introduce quite a lot of boilerplate wehn we could just access data in the service - either way has its pros and cons though, so i don't mind too much

Reply:
Yes, that's my thinking, is that the app will never modify the postcodes, it only uses them as reference data, so it seems overkill to have a whole class just to run `.FirstOrDefaultAsync(x => x.Postcode == postcode)`
If we ever modified them, i would hands down agree with this approach, but here I think it's much less clear. Unless this separation would make the service easier to test, but I think we can mock out the context just as easily?

Reply:
Agree with your reasoning, I will remove the repository then